File: server/src/main/java/com/project/insurance/service/FraudService.java
package com.project.insurance.service;

import java.util.List;

import com.project.insurance.dto.response.*;

public interface FraudService {

    FraudAlertResponseDTO analyzeClaim(Long claimId);

    FraudAlertResponseDTO markFraud(Long alertId, boolean fraud);

    List<FraudAlertResponseDTO> getFraudsByClaim(Long claimId);
}

File: client/src/services/fraudService.js
export const analyzeClaim = (api, claimId) => api.post(`/fraud/analyze/${claimId}`).then(r => r.data);
export const getFraudsByClaim = (api, claimId) => api.get(`/fraud/claim/${claimId}`).then(r => r.data);
export const markFraud = (api, alertId, fraud) => api.put(`/fraud/${alertId}`, null, { params: { fraud } }).then(r => r.data);